src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -24,20 +25,22 @@ var {
   ClassComponent,
   HostRoot,
 } = require('ReactTypeOfWork');
+const {
+  createCursor,
+  pop,
+  push,
+  reset,
+} = require('ReactFiberStack');
 
 if (__DEV__) {
   var checkReactTypeSpec = require('checkReactTypeSpec');
 }
 
-let index = -1;
-const contextStack : Array<Object> = [];
-const didPerformWorkStack : Array<boolean> = [];
+let contextStackCursor : StackCursor<?Object> = createCursor((null: ?Object));
+let didPerformWorkStackCursor : StackCursor<?boolean> = createCursor((null: ?boolean));
```

Comment:
If you set the default here to `(false : boolean)` you can get rid of the nullable type.